### PR TITLE
feat: Preview images on `/code-checker`

### LINKED ISSUE #10207: Preview images on `/code-checker`
The other drops so far look fine. To prevent this issue in the future, can we put an image preview with capture workers under `/code-checker` for now?

_Originally posted by @preschian in https://github.com/kodadot/nft-gallery/issues/10144#issuecomment-2080369550_

### PR DESCRIPTION
**Pull request by hassnian**
## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

## Context

❗️❗️ Needs https://github.com/kodadot/private-workers/pull/154
- [x] Closes #10207

## Needs QA check

- @kodadot/qa-guild please review

#### Did your issue had any of the "$" label on it?

- [x] Fill up your DOT address: [Payout](https://kodadot.xyz/dot/transfer?target=16faLfsywwNATaEfbH2ah75dn6ZmctQWpMS5G4KFhbmj5hnD)

## Screenshot 📸

- [x] My fix has changed **something** on UI;

### CODE REVIEW

**Comment by hassnian** [comment]
@exezbcz I need a state for failed previews , thanks

**Review by Jarsen136 — APPROVED** [review]

File: components/codeChecker/massPreview/Grid.vue [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      class="relative w-full aspect-square">
+      <NeoSkeleton
+        v-if="item.loading"
+        class="border border-border-color overflow-hidden absolute h-full"
+        width="100%"
+        height="100%"
+        :rounded="false"
+        no-margin />
+
+      <slot :item="item" />
```
Comment (daiagi) [inline components/codeChecker/massPreview/Grid.vue:15]: so, it seems that item  contains more then just loading

isn't it overly complicated?

pass the loading array to this Grid wrapper
and the items directly to BaeMediaItem and to CodeCheckerSandboxIFrame

File: components/codeChecker/massPreview/Canvas.vue [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+})
+
+watch(
+  [active, () => props.assets],
+  ([active]) => {
+    if (active) {
+      generateMassPreview()
+    }
+  },
+  { immediate: true },
```
Comment (daiagi) [inline components/codeChecker/massPreview/Canvas.vue:76]: if active satrts as false, and get's switched on by user
why need immediate?

File: components/codeChecker/massPreview/CodeCheckerMassPreview.vue [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    <CodeCheckerMassPreviewCapture
+      :assets="assets"
+      :index-content="indexContent"
+      :previews="PREVIEWS_AMOUNT" />
+  </div>
+</template>
+<script lang="ts" setup>
+import { AssetMessage } from '../types'
+
+const PREVIEWS_AMOUNT = 12
```
Comment (daiagi) [inline components/codeChecker/massPreview/CodeCheckerMassPreview.vue:16]: kinda hidden away, and very likely to get changed in the future / by user

i.e make it props with default value

**Review by daiagi — CHANGES_REQUESTED** [review]

File: components/codeChecker/massPreview/Grid.vue (hunk 1 above)
Comment (hassnian) [reply components/codeChecker/massPreview/Grid.vue]: done

File: components/codeChecker/CodeChecker.vue [hunk 4]
```diff
@@ -143,13 +143,31 @@
       class="w-full lg:w-1/2 flex flex-col items-center lg:mt-4 lg:items-end">
       <!-- Content of the second column -->
       <CodeCheckerPreviewCard
+        ref="codeCheckerPreview"
```
[the remark this answers is not in the document]
Comment (hassnian) [reply components/codeChecker/CodeChecker.vue]: done

File: services/playground.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
+import { $fetch } from 'ofetch'
+
+const BASE_URL =
+  window.location.host === 'kodadot.xyz'
+    ? 'https://playground.kodadot.workers.dev/'
+    : 'https://playground-beta.kodadot.workers.dev/'
+
+const PUBLIC_R2_BUCKET_URL = 'ACTUAL_URL.r2.dev'
```
[the remark this answers is not in the document]
Comment (hassnian) [reply services/playground.ts]: was wating for the worker the be merged , [new code](https://github.com/kodadot/nft-gallery/blob/58553eacafd5688f445620820222e58b1bd93e2b/services/playground.ts#L7-L9)

File: components/codeChecker/massPreview/Canvas.vue (hunk 2 above)
Comment (hassnian) [reply components/codeChecker/massPreview/Canvas.vue]: nice catch, forgot to update this after refactoring the code, fixed

**Review by daiagi — APPROVED** [review]

**Comment by prury** [comment]
well, i get some errors sometimes when use the capturing service test, but i guess is related to the art uploaded, will keep an eye on it, merging.

@preschian let's say a new art is uploaded and the capturing service is somehow not showing the images properly, what's the course of action? inform devs?

**Pull request merged**